src/test/java/reposense/parser/ArgsParserTest.java

```diff
@@ -0,0 +1,177 @@
+package reposense.parser;
+
+import java.io.File;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.Optional;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import net.sourceforge.argparse4j.inf.ArgumentParserException;
+
+import reposense.frontend.CliArguments;
+
+public class ArgsParserTest {
+
+    private String projectDirectory;
+
+    public ArgsParserTest() {
+        projectDirectory = System.getProperty("user.dir");
+    }
+
+    private Date getDate(int year, int month, int date) {
+        return new Calendar.Builder()
+                            .setDate(year, month, date)
+                            .setTimeOfDay(0, 0, 0)
+                            .build()
+                            .getTime();
+    }
+
+    @Test
+    public void allCorrectInputs_success() throws ArgumentParserException {
+        Path configFilePath = Paths.get(projectDirectory + "/sample_small.csv");
+        Path outputDirectoryPath = Paths.get(projectDirectory);
+        String input = String.format("-config %s -output %s -since 01/07/2017 -until 30/11/2017",
+                configFilePath, outputDirectoryPath);
+        String[] args = input.split(" ");
+
+        CliArguments cliArguments = ArgsParser.parse(args);
+        Assert.assertEquals(configFilePath.toAbsolutePath(), cliArguments.getConfigFilePath().toAbsolutePath());
+        Assert.assertEquals(outputDirectoryPath.toAbsolutePath(), cliArguments.getTargetFilePath().toAbsolutePath());
+
+        Date expectedSinceDate = getDate(2017, Calendar.JULY, 1);
+        Date expectedUntilDate = getDate(2017, Calendar.NOVEMBER, 30);
+        Assert.assertEquals(expectedSinceDate, cliArguments.getSinceDate().get());
+        Assert.assertEquals(expectedUntilDate, cliArguments.getUntilDate().get());
+    }
+
+    @Test
+    public void absolutePath_toConfigFile_success() throws ArgumentParserException {
+        String directory = Paths.get(projectDirectory + "/sample_small.csv").toAbsolutePath().toString();
+        String input = String.format("-config %s", directory);
+        String[] args = input.split(" ");
+        ArgsParser.parse(args);
+    }
+
+    @Test
+    public void relativePath_toConfigFile_success() throws ArgumentParserException {
+        String input = String.format("-config %s", "sample_small.csv");
+        String[] args = input.split(" ");
+        ArgsParser.parse(args);
+    }
+
+    @Test
+    public void absolutePath_toOutputDirectory_success() throws ArgumentParserException {
```

Comment:
`parse_configFileAndOutputDirectory_success()`
same as above, we can mix the `absolutePath` and `relativePath` test here together